tasks/model.go

```diff
@@ -77,3 +94,80 @@ var statusNames = map[Status]string{
 func (s Status) String() string {
 	return statusNames[s]
 }
+
+// ConnectivityStages are stages that occur prior to initiating a deal
+var ConnectivityStages = map[string]StageData{
+	"MinerOnline": {
+		Description:      "Miner is online",
+		ExpectedDuration: "a few seconds",
+	},
+	"QueryAsk": {
+		Description:      "Miner responds to query ask",
+		ExpectedDuration: "a few seconds",
+	},
+	"CheckPrice": {
+		Description:      "Miner meets price criteria",
+		ExpectedDuration: "",
+	},
+	"ClientImport": {
+		Description:      "Importing data into Lotus",
+		ExpectedDuration: "a few minutes",
+	},
+}
+
+// RetrievalStages are stages that occur in a retrieval deal
+var RetrievalStages = map[string]StageData{
+	"ProposeRetrieval": {
+		Description:      "Send retrieval to miner",
+		ExpectedDuration: "",
+	},
+	"DealAccepted": {
+		Description:      "Miner accepts deal",
+		ExpectedDuration: "a few seconds",
+	},
+	"FirstByteReceived": {
+		Description:      "First byte of data received from miner",
+		ExpectedDuration: "a few seconds, or several hours when unsealing",
+	},
+	"DealComplete": {
+		Description:      "All bytes received and deal is completed",
+		ExpectedDuration: "a few seconds",
+	},
+}
```

Comment:
When a deal is scheduled to run in the future but is not yet running, do we want a stage to indicate this?  Maybe having a NULL `StageName` for an InProgress task is enough?

Is there any way a deal can be cancelled before it is complete?

Reply:
yea null = in progress but not running. Also this will be a subsecond amount of time.